engine/dlib/src/dlib/configfile.cpp

```diff
@@ -611,9 +688,30 @@ namespace dmConfigFile
         return default_value;
     }
 
-    int32_t GetInt(HConfig config, const char* key, int32_t default_value)
+    static bool GetStringOverride(HConfig config, const char* key, const char* default_value, const char** out_value)
     {
-        const char* tmp = GetString(config, key, 0);
+        const PluginDesc* plugin = GetFirstExtension();
+        while (plugin != 0)
+        {
+            if (plugin->m_GetString && plugin->m_GetString(config, key, default_value, out_value))
+                return true;
+            plugin = plugin->m_Next;
+        }
+        return false;
+    }
+
+    const char* GetString(HConfig config, const char* key, const char* default_value)
+    {
+        const char* base_value = GetBaseString(config, key, default_value);
+        const char* out_value = 0;
+        if (GetStringOverride(config, key, base_value, &out_value))
+            return out_value;
+        return base_value;
```

Comment:
The flow I settled on is that we get the original value from the config (or the default one if it didn't exist).
Then I call the extension with the key and the original value.

This allows the extension to e.g. decode a confuscated string.